main.cpp

```diff
@@ -27,6 +28,11 @@ void paxos::startServer() {
     });
 }
 
+void paxos::startBatchers() {
+    int i = 0;
+    participants.emplace_back(std::thread([i]{batcher {i};}));
```

Comment:
I think you'll be able to add f+1 batchers and change nothing else and have things work